lib/webmachine.ml

```diff
@@ -103,9 +103,13 @@ module type S = sig
   type 'body provider = ('body, 'body) op
   type 'body acceptor = (bool, 'body) op
 
+  type www_authenticate =
+    { scheme : string; realm : string; params : (string * string) list }
+
   type auth =
     [ `Authorized
     | `Basic of string
+    | `Unauthorized of www_authenticate
```

Comment:
`Challenge` is a more appropriate name for this variant, as it's the names used in [RFC7235](https://tools.ietf.org/html/rfc7235#section-4.1)